x-pack/plugins/actions/server/lib/get_execution_log_aggregation.ts

```diff
@@ -0,0 +1,446 @@
+      aggs: {
+        executionUuid: {
+          // Bucket by execution UUID
+          terms: {
+            field: EXECUTION_UUID_FIELD,
+            size: DEFAULT_MAX_KPI_BUCKETS_LIMIT,
+            order: formatSortForTermSort([{ timestamp: { order: 'desc' } }]),
+          },
+          aggs: {
+            executionUuidSorted: {
+              bucket_sort: {
+                from: 0,
+                size: DEFAULT_MAX_KPI_BUCKETS_LIMIT,
+                gap_policy: 'insert_zeros' as estypes.AggregationsGapPolicy,
+              },
+            },
+            actionExecution: {
+              filter: {
+                bool: {
+                  must: [
+                    {
+                      match: {
+                        [ACTION_FIELD]: 'execute',
+                      },
+                    },
+                  ],
+                },
+              },
+              aggs: {
+                executeStartTime: {
+                  min: {
+                    field: START_FIELD,
+                  },
+                },
+                actionExecutionOutcomes: {
+                  terms: {
+                    size: 3,
+                    field: OUTCOME_FIELD,
+                  },
+                },
+              },
+            },
+            minExecutionUuidBucket: {
+              bucket_selector: {
+                buckets_path: {
+                  count: 'actionExecution._count',
+                },
+                script: {
+                  source: 'params.count > 0',
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+  };
+};
+
+export function getExecutionLogAggregation({
+  filter,
+  page,
+  perPage,
+  sort,
+}: IExecutionLogAggOptions) {
+  // Check if valid sort fields
+  const sortFields = flatMap(sort as estypes.SortCombinations[], (s) => Object.keys(s));
+  for (const field of sortFields) {
+    if (!Object.keys(ExecutionLogSortFields).includes(field)) {
+      throw Boom.badRequest(
+        `Invalid sort field "${field}" - must be one of [${Object.keys(ExecutionLogSortFields).join(
+          ','
+        )}]`
+      );
+    }
+  }
+
+  // Check if valid page value
+  if (page <= 0) {
+    throw Boom.badRequest(`Invalid page field "${page}" - must be greater than 0`);
+  }
+
+  // Check if valid page value
+  if (perPage <= 0) {
+    throw Boom.badRequest(`Invalid perPage field "${perPage}" - must be greater than 0`);
+  }
+
+  const dslFilterQuery: estypes.QueryDslBoolQuery['filter'] = buildDslFilterQuery(filter);
+
+  return {
+    executionLogAgg: {
+      ...(dslFilterQuery ? { filter: { bool: { filter: dslFilterQuery } } } : {}),
+      aggs: {
+        // Get total number of executions
+        executionUuidCardinality: {
+          filter: {
+            bool: {
+              must: [
+                {
+                  match: {
+                    [ACTION_FIELD]: 'execute',
```

Comment:
Since we're doing a top level provider filter (that also filters out `execute-start`), these aggregations should only run on the actions docs, so no rule provider docs should be aggregated over.

Reply:
I think we're good then, thx.  I think my original concern was that we'd be also looking through the `alerting` docs, but that's filtered out at the client level, so that's already handled.